kevm-pyk/src/kevm_pyk/__main__.py

```diff
@@ -71,144 +88,214 @@ def _ignore_arg(args: dict[str, Any], arg: str, cli_option: str) -> None:
 
 def main() -> None:
     sys.setrecursionlimit(15000000)
-    parser = _create_argument_parser()
+
+    kevm_cli = CLI(
+        (
+            KastCommand,
+            KompileSpecCommand,
+            ProveLegacyCommand,
+            ProveCommand,
+            PruneProofCommand,
+            RunCommand,
+            ShowKCFGCommand,
+            VersionCommand,
+            ViewKCFGCommand,
+        )
+    )
+    parser = kevm_cli.create_argument_parser()
     args = parser.parse_args()
+
     logging.basicConfig(level=_loglevel(args), format=_LOG_FORMAT)
+    command = kevm_cli.generate_command({key: val for (key, val) in vars(args).items() if val is not None})
+    command.exec()
```

Comment:
I'm curious about the performance of this approach. Suppose we make it so that `KastCommand` is a no-op (take out the actual implementation in teh `exec()` method), how long does running `kevm-pyk kast ....` take? Basically, what is the overhead associated with this method of putting together the CLI?

Reply:
As far as I can tell, any overhead is negligible. With or without a no-op `exec()`, and on master as well, it takes between 0.21 and 0.23 seconds, and the vast majority of that must be loading the code or something, because it fails to go down below that even when I exit right at the start of `main()`.

Is there a particular reason we'd expect a performance hit from code like this?

Reply:
Just more complicated class structure that Python has to load, but I guess that's neglible compared to the overhead of just starting the python process up.

Reply:
Thanks for checking!